openmc/data/endf.c

```diff
@@ -0,0 +1,39 @@
+#include <stdlib.h>
+
+double cfloat_endf(const char* buffer, int n)
+{
+  char arr[12]; // 11 characters plus a null terminator
+  int j = 0; // current position in arr
+  int found_significand = 0;
+  int found_exponent = 0;
+  for (int i = 0; i < n; ++i) {
```

Comment:
Note that there is no bounds checking performed on `n`, and so therefore `i`. Therefore, `arr` could be overflowed. If the input to this function can be customized by a user by using a deliberately crafted input value then at the very least a stack corruption could be triggered and cause the program to crash on return, or a security issue could be potentially exploited.